include/swift/AST/ASTContext.h

```diff
@@ -863,8 +863,13 @@ class ASTContext {
   /// like `<T>`.
   CanGenericSignature getSingleGenericParameterSignature() const;
   
+  /// Whether our effective Swift version is less than the given version
+  bool isSwiftVersionLessThan(unsigned major, unsigned minor = 0) const {
+    return LangOpts.isSwiftVersionLessThan(major, minor);
+  }
+
   /// Whether our effective Swift version is in the Swift 3 family
-  bool isSwiftVersion3() const { return LangOpts.isSwiftVersion3(); }
+  bool isSwiftVersion3() const { return isSwiftVersionLessThan(4); }
```

Comment:
Definitely an improvement, but I still don't like the name `isSwiftVersion3()`; `isSwiftVersionLessThan(4)` or `< 4` would be far clearer, IMO.

Reply:
@davezarzycki I was wondering if you were still working on adding this to Swift in another pull request? I need to conditionally act on the Swift mode (4 or 5) for another fix I'm working on. Thanks

Reply:
Hi @gspiers. I'm am not. This should have been a very trivial change, but after spending too many hours trying to debug a regression from this change in the Swift 3 to 4 migrator, I've handed off the work to @ematejska.

Reply:
Thanks @davezarzycki for the quick reply. I'll keep an eye out for the changes to land in master before I submit my fix.